### ISSUE TITLE
train log

### ISSUE THREAD

**Issue by taofuyu** [supperted825/FairMOT-X#7]
Thanks for your work!
I follow your code and then implement yolox+fairmot in mmdetection. But the ReID loss does not  descend.
Would you mind uploading your train log as a reference ?

**Comment by supperted825** [comment]
Hi there @taofuyu, sorry for the late reply. I have attached an example training log for the YOLOX-L variant with uncertainty loss. The reID loss should indeed descend.

```
2022-01-30-23-56: epoch: 1 |tot_loss 1.275687 | iou_loss 1.667313 | l1_loss 0.000000 | conf_loss 0.790379 | cls_loss 0.542091 | reid_loss 0.161237 | time 714.966667
2022-01-31-23-26: epoch: 2 |tot_loss 0.969020 | iou_loss 1.563289 | l1_loss 0.000000 | conf_loss 0.650327 | cls_loss 0.487000 | reid_loss 0.132141 | time 666.683333
2022-02-01-10-28: epoch: 3 |tot_loss 0.833890 | iou_loss 1.514289 | l1_loss 0.000000 | conf_loss 0.587463 | cls_loss 0.464571 | reid_loss 0.121439 | time 661.116667
2022-02-01-21-27: epoch: 4 |tot_loss 0.729266 | iou_loss 1.480873 | l1_loss 0.000000 | conf_loss 0.548227 | cls_loss 0.451540 | reid_loss 0.113156 | time 658.716667
2022-02-02-08-20: epoch: 5 |tot_loss 0.617933 | iou_loss 1.454842 | l1_loss 0.000000 | conf_loss 0.519035 | cls_loss 0.442130 | reid_loss 0.103950 | time 651.650000
2022-02-02-19-14: epoch: 6 |tot_loss 0.497730 | iou_loss 1.434974 | l1_loss 0.000000 | conf_loss 0.498067 | cls_loss 0.435463 | reid_loss 0.094039 | time 653.550000
2022-02-03-06-19: epoch: 7 |tot_loss 0.362815 | iou_loss 1.417393 | l1_loss 0.000000 | conf_loss 0.479551 | cls_loss 0.429775 | reid_loss 0.083651 | time 664.100000
2022-02-03-23-40: epoch: 8 |tot_loss 0.219405 | iou_loss 1.402467 | l1_loss 0.000000 | conf_loss 0.464665 | cls_loss 0.425286 | reid_loss 0.073569 | time 679.050000
2022-02-04-12-16: epoch: 9 |tot_loss 0.070405 | iou_loss 1.389431 | l1_loss 0.000000 | conf_loss 0.451879 | cls_loss 0.421334 | reid_loss 0.064219 | time 675.433333
2022-02-04-23-26: epoch: 10 |tot_loss -0.080065 | iou_loss 1.378529 | l1_loss 0.000000 | conf_loss 0.441913 | cls_loss 0.418106 | reid_loss 0.055845 | time 669.450000
2022-02-05-10-36: epoch: 11 |tot_loss -0.231108 | iou_loss 1.368566 | l1_loss 0.000000 | conf_loss 0.432812 | cls_loss 0.415275 | reid_loss 0.048493 | time 668.416667
2022-02-05-21-50: epoch: 12 |tot_loss -0.380542 | iou_loss 1.360556 | l1_loss 0.000000 | conf_loss 0.425717 | cls_loss 0.412891 | reid_loss 0.042094 | time 673.733333
2022-02-06-08-48: epoch: 13 |tot_loss -0.527605 | iou_loss 1.351162 | l1_loss 0.000000 | conf_loss 0.417726 | cls_loss 0.410439 | reid_loss 0.036666 | time 656.733333
2022-02-06-19-47: epoch: 14 |tot_loss -0.669063 | iou_loss 1.344112 | l1_loss 0.000000 | conf_loss 0.412340 | cls_loss 0.408535 | reid_loss 0.032038 | time 658.416667
2022-02-12-04-22: epoch: 15 |tot_loss -0.805410 | iou_loss 1.338777 | l1_loss 0.000000 | conf_loss 0.407129 | cls_loss 0.406966 | reid_loss 0.028110 | time 796.516667
2022-02-12-11-58: epoch: 16 |tot_loss 0.476022 | iou_loss 1.352580 | l1_loss 0.403633 | conf_loss 0.534101 | cls_loss 0.419759 | reid_loss 0.063736 | time 453.050000
2022-02-12-19-42: epoch: 17 |tot_loss 0.079342 | iou_loss 1.308309 | l1_loss 0.385502 | conf_loss 0.477429 | cls_loss 0.405772 | reid_loss 0.056545 | time 461.500000
2022-02-13-03-13: epoch: 18 |tot_loss -0.035285 | iou_loss 1.289889 | l1_loss 0.375805 | conf_loss 0.457013 | cls_loss 0.400698 | reid_loss 0.051778 | time 447.966667
2022-02-13-10-53: epoch: 19 |tot_loss -0.138405 | iou_loss 1.274930 | l1_loss 0.369788 | conf_loss 0.440199 | cls_loss 0.396391 | reid_loss 0.047494 | time 457.683333
2022-02-13-18-26: epoch: 20 |tot_loss -0.233669 | iou_loss 1.261977 | l1_loss 0.365125 | conf_loss 0.428098 | cls_loss 0.392921 | reid_loss 0.043765 | time 452.033333
```

**Comment by taofuyu** [comment]
> Hi there @taofuyu, sorry for the late reply. I have attached an example training log for the YOLOX-L variant with uncertainty loss. The reID loss should indeed descend.
> 
> ```
> 2022-01-30-23-56: epoch: 1 |tot_loss 1.275687 | iou_loss 1.667313 | l1_loss 0.000000 | conf_loss 0.790379 | cls_loss 0.542091 | reid_loss 0.161237 | time 714.966667
> 2022-01-31-23-26: epoch: 2 |tot_loss 0.969020 | iou_loss 1.563289 | l1_loss 0.000000 | conf_loss 0.650327 | cls_loss 0.487000 | reid_loss 0.132141 | time 666.683333
> 2022-02-01-10-28: epoch: 3 |tot_loss 0.833890 | iou_loss 1.514289 | l1_loss 0.000000 | conf_loss 0.587463 | cls_loss 0.464571 | reid_loss 0.121439 | time 661.116667
> 2022-02-01-21-27: epoch: 4 |tot_loss 0.729266 | iou_loss 1.480873 | l1_loss 0.000000 | conf_loss 0.548227 | cls_loss 0.451540 | reid_loss 0.113156 | time 658.716667
> 2022-02-02-08-20: epoch: 5 |tot_loss 0.617933 | iou_loss 1.454842 | l1_loss 0.000000 | conf_loss 0.519035 | cls_loss 0.442130 | reid_loss 0.103950 | time 651.650000
> 2022-02-02-19-14: epoch: 6 |tot_loss 0.497730 | iou_loss 1.434974 | l1_loss 0.000000 | conf_loss 0.498067 | cls_loss 0.435463 | reid_loss 0.094039 | time 653.550000
> 2022-02-03-06-19: epoch: 7 |tot_loss 0.362815 | iou_loss 1.417393 | l1_loss 0.000000 | conf_loss 0.479551 | cls_loss 0.429775 | reid_loss 0.083651 | time 664.100000
> 2022-02-03-23-40: epoch: 8 |tot_loss 0.219405 | iou_loss 1.402467 | l1_loss 0.000000 | conf_loss 0.464665 | cls_loss 0.425286 | reid_loss 0.073569 | time 679.050000
> 2022-02-04-12-16: epoch: 9 |tot_loss 0.070405 | iou_loss 1.389431 | l1_loss 0.000000 | conf_loss 0.451879 | cls_loss 0.421334 | reid_loss 0.064219 | time 675.433333
> 2022-02-04-23-26: epoch: 10 |tot_loss -0.080065 | iou_loss 1.378529 | l1_loss 0.000000 | conf_loss 0.441913 | cls_loss 0.418106 | reid_loss 0.055845 | time 669.450000
> 2022-02-05-10-36: epoch: 11 |tot_loss -0.231108 | iou_loss 1.368566 | l1_loss 0.000000 | conf_loss 0.432812 | cls_loss 0.415275 | reid_loss 0.048493 | time 668.416667
> 2022-02-05-21-50: epoch: 12 |tot_loss -0.380542 | iou_loss 1.360556 | l1_loss 0.000000 | conf_loss 0.425717 | cls_loss 0.412891 | reid_loss 0.042094 | time 673.733333
> 2022-02-06-08-48: epoch: 13 |tot_loss -0.527605 | iou_loss 1.351162 | l1_loss 0.000000 | conf_loss 0.417726 | cls_loss 0.410439 | reid_loss 0.036666 | time 656.733333
> 2022-02-06-19-47: epoch: 14 |tot_loss -0.669063 | iou_loss 1.344112 | l1_loss 0.000000 | conf_loss 0.412340 | cls_loss 0.408535 | reid_loss 0.032038 | time 658.416667
> 2022-02-12-04-22: epoch: 15 |tot_loss -0.805410 | iou_loss 1.338777 | l1_loss 0.000000 | conf_loss 0.407129 | cls_loss 0.406966 | reid_loss 0.028110 | time 796.516667
> 2022-02-12-11-58: epoch: 16 |tot_loss 0.476022 | iou_loss 1.352580 | l1_loss 0.403633 | conf_loss 0.534101 | cls_loss 0.419759 | reid_loss 0.063736 | time 453.050000
> 2022-02-12-19-42: epoch: 17 |tot_loss 0.079342 | iou_loss 1.308309 | l1_loss 0.385502 | conf_loss 0.477429 | cls_loss 0.405772 | reid_loss 0.056545 | time 461.500000
> 2022-02-13-03-13: epoch: 18 |tot_loss -0.035285 | iou_loss 1.289889 | l1_loss 0.375805 | conf_loss 0.457013 | cls_loss 0.400698 | reid_loss 0.051778 | time 447.966667
> 2022-02-13-10-53: epoch: 19 |tot_loss -0.138405 | iou_loss 1.274930 | l1_loss 0.369788 | conf_loss 0.440199 | cls_loss 0.396391 | reid_loss 0.047494 | time 457.683333
> 2022-02-13-18-26: epoch: 20 |tot_loss -0.233669 | iou_loss 1.261977 | l1_loss 0.365125 | conf_loss 0.428098 | cls_loss 0.392921 | reid_loss 0.043765 | time 452.033333
> ```

Thanks ! Another question is, how do you construct your batch ? i.e., should I construct a  batch like "batch-size=32, 4 videos and 8 images per video" ?

**Comment by starhou** [comment]
My reid_loss is always 0.
![image](https://user-images.githubusercontent.com/46232202/176144556-818d5cda-4f65-44e5-889e-f2ddb0caed55.png)

**Comment by youonlytrackonce** [comment]
> 

Hello,

would you please share your hardware configuration and conda env specifications? (rtx 3090, cuda 11.6, pytorch 1.8.2, torchvision etc etc)

Thank you!

**Comment by youonlytrackonce** [comment]
> My reid_loss is always 0. ![image](https://user-images.githubusercontent.com/46232202/176144556-818d5cda-4f65-44e5-889e-f2ddb0caed55.png)

Hello, my tot_loss is also always negative. Did you solve your problem?
